### ISSUE TITLE
Request: Recurring Sub-Tasks add the Ebbinghaus forgetting curve

### ISSUE THREAD

**Issue by hikerun** [mhoeher/opentodolist#92]
Hello, is it possible to add an Ebbinghaus forgetting curve to the recurring Sub-Tasks, for example, the repeat date set to 1,2,4,7,15 days.

**Comment by mhoeher** [comment]
Hi @hikerun,

generally, as we're talking about software, a lot is possible 😉

I have to admit I heard about the Ebbinghaus forgetting curve the very first time. I found e.g. this article describing it a bit: https://www.mindtools.com/a9wjrjw/ebbinghauss-forgetting-curve

I guess that especially the part about "spaced learning" is what you had in mind when opening the issue, correct?

I could imagine that implementing something like this is easily possible:

- OpenTodoList already supports setting a repetition pattern, e.g. you can repeat a task every N days.
- What would be possible is that the user could also set up a "factor" by which this number N is tweaked in each iteration. For example, if set to 2, a repeating task would recur after 1, 2, 4, 8, 16, ... days.

Is this what you roughly had in mind?

What I'd be interested in: Do you have any other use case for such a recurrence pattern? From the linked article, I would assume that this could be useful for learning. Do you see any other use cases?

**Comment by hikerun** [comment]
Yes, it could be useful for learning. For example, I can use it to remind me to review an article I've read. I can review it on day 1, day 2, day 4, day 7, and day 15 to help me memorize it better.

I think set up a "factor" is a good way.

**Comment by real-easy** [comment]
i think it is also an cool feature for new installations (hardware), in first time we can check it often, than later we check it but less.
but i think we must define a min (this can be the intervall value) and a max value, only an factor may be to less.
and also an preview of the intervall (numbers day / weeks or so on) will be helpfull.

**Comment by mhoeher** [comment]
Thanks the two of you!

Okay, I guess it makes sense to support this. I'll created a tracking issue over on GitLab for the implementation: https://gitlab.com/rpdev/opentodolist/-/issues/632

Feel free to comment there as well if you have something to add. I think this feature should be relatively easy to add, so might be in for the next release (but please don't take this for granted 😉).
